File: WebClient/src/app/model/step.ts
import { IIrrigationAction } from "./irrigationaction";

export class IStep {
    constructor(
        public id: number,
        public Sequence: number,
        public Duration: number,
        public SolenoidId: number,
        public SolenoidName: string,
        public RequiresPump: boolean,
        public ProgramId: number,
        public IrrigationActionId: number,
        public IrrigationAction: IIrrigationAction,
        public createdAt: Date,
        public updatedAt: Date
    ) {}
}

// const step = new IStep(1, 1, 60, 1, 'solenoidName', true, 1, 0, new Date(), new Date());

File: Device/DeviceController.Data/Step.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DeviceController.Data
{
    public class Step
    {
        public int Id;
        public int Sequence;
        public int Duration;
        public int SolenoidId;
        public string SolenoidName;
        public bool RequiresPump;
        public int IrrigationActionId;
        public DateTime CreatedAt;
        public DateTime UpdatedAt;
    }
}
